plugins/dell-dock/fu-dell-dock-status.c

```diff
@@ -103,12 +103,20 @@ fu_dell_dock_status_open(FuDevice *device, GError **error)
 	if (fu_device_get_proxy(device) == NULL)
 		fu_device_set_proxy(device, fu_device_get_parent(device));
 
+	if (fu_device_get_proxy(device) == NULL) {
+		g_set_error_literal(error, FWUPD_ERROR, FWUPD_ERROR_INTERNAL, "no proxy");
+		return FALSE;
+	}
```

Comment:
wait isn't this impossible?  Look at the check above.

Reply:
It's kinda unclear, but the logic is sound -- e.g.

	if (fu_device_get_proxy(device) == NULL)
		fu_device_set_proxy(device, fu_device_get_parent(device));
   //then we check fu_device_get_proxy(device)

It's certainly a bit weird. I'll fix it up.

Reply:
Fixed.